src/protocol/mod.rs

```diff
@@ -332,6 +334,11 @@ impl WebSocketContext {
         Stream: Read + Write,
     {
         if let Some(mut frame) = self.frame.read_frame(stream, self.config.max_frame_size)? {
+
+            if !self.state.can_read() {
+                return Err(Error::Protocol("Remote sent frame after having sent a Close Frame".into()));
+            }
```

Comment:
@application-developer-DA   Ok, this was right after all. Since it's conditional on `can_read`, it means that the we already got the acknowledgement in question, so really, we shouldn't be doing any processing in this block any more if `can_read` returns false.

The only reason the method needs to be called at all is to detect if the remote closed the underlying stream. I will update the PR back to what it was.